### PR TITLE
LinkedList [C#]

### LINKED ISSUE #474: LinkedList [C#]
#### This issue is about [Linked List](https://en.wikipedia.org/wiki/Linked_list).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by ransty**
Fixes #474

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by ransty** [comment]
The check error was so simple (the compiler clearly told me what it was), however it took me over an hour to figure out.

File: breadth_first_search/BreadthFirstSearch.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            if node not in seen:
+                seen.add(node)
+                queue.append(node)
+    print(returnprint)
+
+def visit(n):
+    print(n)
+
+if __name__ == '__main__':
+    graph = {0: [1, 2], 1: [2, 0], 2: []}
```
Comment (singhpratyush) [inline breadth_first_search/BreadthFirstSearch.py]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `python spacing`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
+++ b/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
@@ -16,7 +16,7 @@
     print(n)
 
 if __name__ == '__main__':
-    graph = {0: [1, 2], 1: [2, 0], 2: []} 
+    graph = {0: [1, 2], 1: [2, 0], 2: []}
     bfs(graph, 1) # 1 2 0
 
     graph2 = {0: [1, 3], 1: [3, 2], 2: [2, 1], 3: []}
```

File: breadth_first_search/BreadthFirstSearch.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import collections
```
Comment (singhpratyush) [inline breadth_first_search/BreadthFirstSearch.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
+++ b/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
@@ -1,4 +1,5 @@
 import collections
+
 
 def bfs(graph, root):
     returnprint = 'BFS Order: '
```

File: breadth_first_search/BreadthFirstSearch.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    seen, queue = set([root]), collections.deque([root])
+    while queue:
+        vertex = queue.popleft()
+        returnprint += str(vertex) + '  '
+        for node in graph[vertex]:
+            if node not in seen:
+                seen.add(node)
+                queue.append(node)
+    print(returnprint)
+
```
Comment (singhpratyush) [inline breadth_first_search/BreadthFirstSearch.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
+++ b/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
@@ -11,6 +11,7 @@
                 seen.add(node)
                 queue.append(node)
     print(returnprint)
+
 
 def visit(n):
     print(n)
```

File: breadth_first_search/BreadthFirstSearch.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+def visit(n):
+    print(n)
+
```
Comment (singhpratyush) [inline breadth_first_search/BreadthFirstSearch.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
+++ b/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
@@ -15,9 +15,10 @@
 def visit(n):
     print(n)
 
+
 if __name__ == '__main__':
-    graph = {0: [1, 2], 1: [2, 0], 2: []} 
-    bfs(graph, 1) # 1 2 0
+    graph = {0: [1, 2], 1: [2, 0], 2: []}
+    bfs(graph, 1)  # 1 2 0
 
     graph2 = {0: [1, 3], 1: [3, 2], 2: [2, 1], 3: []}
     bfs(graph2, 2) # 2 1 3
```

File: breadth_first_search/BreadthFirstSearch.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+if __name__ == '__main__':
+    graph = {0: [1, 2], 1: [2, 0], 2: []} 
+    bfs(graph, 1) # 1 2 0
+
+    graph2 = {0: [1, 3], 1: [3, 2], 2: [2, 1], 3: []}
+    bfs(graph2, 2) # 2 1 3
```
Comment (singhpratyush) [inline breadth_first_search/BreadthFirstSearch.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
+++ b/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
@@ -20,7 +20,7 @@
     bfs(graph, 1) # 1 2 0
 
     graph2 = {0: [1, 3], 1: [3, 2], 2: [2, 1], 3: []}
-    bfs(graph2, 2) # 2 1 3
+    bfs(graph2, 2)  # 2 1 3
 
     graph3 = {0: [1, 2], 1: [2, 3], 2: [3, 4], 3: [4, 0], 4: []}
     bfs(graph3, 1) # 1 2 3 4 0
```

File: breadth_first_search/BreadthFirstSearch.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    graph3 = {0: [1, 2], 1: [2, 3], 2: [3, 4], 3: [4, 0], 4: []}
+    bfs(graph3, 1) # 1 2 3 4 0
```
Comment (singhpratyush) [inline breadth_first_search/BreadthFirstSearch.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
+++ b/tmp/tmpisevponc/breadth_first_search/BreadthFirstSearch.py
@@ -23,9 +23,10 @@
     bfs(graph2, 2) # 2 1 3
 
     graph3 = {0: [1, 2], 1: [2, 3], 2: [3, 4], 3: [4, 0], 4: []}
-    bfs(graph3, 1) # 1 2 3 4 0
+    bfs(graph3, 1)  # 1 2 3 4 0
 
-    graph4 = {0: [1, 3], 1: [3, 2], 2: [2, 4], 3: [4, 3], 4: [4, 0], 5: [0, 1], 6: []}
-    bfs(graph4, 0) # 0 1 3 2 4
+    graph4 = {0: [1, 3], 1: [3, 2], 2: [2, 4],
+              3: [4, 3], 4: [4, 0], 5: [0, 1], 6: []}
+    bfs(graph4, 0)  # 0 1 3 2 4
 
-    bfs(graph4, 2) # 2 4 0 1 3
+    bfs(graph4, 2)  # 2 4 0 1 3
```

File: linked_list/LinkedList.cs [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+
+    private Node<T> head;
+    public int size = 0;
+
+    public LinkedList()
+    {
+        this.head = null;
+    }
+
+    public void addFront(T obj)
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.
ref: https://msdn.microsoft.com/en-us/library/4df752aw(v=vs.71).aspx

File: linked_list/LinkedList.cs [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+    {
+        Node<T> node = new Node<T>(obj);
+        node.next = this.head;
+        this.head = node;
+        this.size++;
+    }
+
+    public void addLast(T obj)
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                temp = temp.next;
+            }
+            Node<T> node = new Node<T>(obj);
+            temp.next = node;
+            size++;
+        }
+    }
+
+    public void add(T obj, int index)
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+                temp = temp.next;
+            }
+            Node<T> node = new Node<T>(obj);
+            node.next = temp.next;
+            temp.next = node;
+            this.size++;
+        }
+    }
+
+    public T removeFront()
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+        {
+            throw new InvalidOperationException();
+        }
+        T objRemoved = this.head.data;
+        this.head = this.head.next;
+        this.size--;
+        return objRemoved;
+    }
+
+    public T remove(int index)
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+            }
+            T objRemoved = temp.next.data;
+            temp.next = temp.next.next;
+            this.size--;
+            return objRemoved;
+        }
+        return removeFront();
+    }
+
+    public T removeLast()
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+                temp = temp.next;
+            }
+            objRemoved = temp.next.data;
+            temp.next = null;
+        }
+        this.size--;
+        return objRemoved;
+    }
+
+    public int search(T obj)
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+        }
+        return -1;
+    }
+
+    public int getSize()
+    {
+        return this.size;
+    }
+
+    public bool isEmpty()
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

File: linked_list/LinkedList.cs [hunk 15]
```diff
@@ -0,0 +1,240 @@
...
+            }
+            else
+            {
+                temp = temp.next;
+            }
+        }
+        return -1;
+    }
+
+    public int getSize()
```
Comment (mohitkyadav) [inline linked_list/LinkedList.cs]: Please use Pascal Case for Method names.

**Comment by ransty** [comment]
Added Pascal Case :)

**Comment by ransty** [comment]
Yeah GitMate broken :(

**Comment by singhpratyush** [comment]
@ransty: Thanks for the contribution :)

**Pull request merged**